### PR TITLE
fix: update to add robust error handling

### PR DESCRIPTION
**Pull request by utsabc**
## Description of the change

- adding consistent error handling to all deletion flows
- returning destination level status to caller
- adding integration names to message to better debug at bugsnag
- adding error response (network failures) to error message for better visibility at bugsnag

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/af/deleteUsers.js [hunk 1]
```diff
@@ -32,7 +46,7 @@ const userDeletionHandler = async (userAttributes, config) => {
   if (!config?.apiToken || !(config?.appleAppId || config?.androidAppId)) {
     throw new ErrorBuilder()
       .setMessage(
-        "API Token and one of Apple ID or Android App Id are required fields for user deletion"
+        "[Appsflyer]::API Token and one of Apple ID or Android App Id are required fields for user deletion"
```
Comment (saikumarrs) [inline src/v0/destinations/af/deleteUsers.js:49]: Remove the prefix, please.

File: src/v0/destinations/braze/deleteUsers.js [hunk 2]
```diff
@@ -43,27 +53,16 @@ const userDeletionHandler = async (userAttributes, config) => {
...
-      }
-      if (!resp || !resp.response) {
-        throw new CustomError("Could not get response", 500);
-      }
-      if (
-        resp &&
-        resp.response &&
-        resp.response.response &&
-        resp.response.response.status !== 200 &&
-        resp.response.response.status !== 404
```
Comment (saikumarrs) [inline src/v0/destinations/braze/deleteUsers.js:61]: Previously, `404` is treated as a success.

File: src/v0/destinations/intercom/deleteUsers.js [hunk 3]
```diff
@@ -25,20 +38,16 @@ const userDeletionHandler = async (userAttributes, config) => {
...
       const resp = await httpSend(requestOptions);
-      if (!resp || !resp.response) {
-        throw new CustomError("Could not get response", 500);
-      }
-      if (
-        resp &&
-        resp.response &&
-        resp.response.response &&
-        resp.response.response.status !== 200 &&
-        resp.response.response.status !== 404 // this will be returned if user is not found. Will send successful to server
```
Comment (saikumarrs) [inline src/v0/destinations/intercom/deleteUsers.js:36]: Previously, `404` is treated as a success.

**Review by saikumarrs — CHANGES_REQUESTED** [review]
Please do not add destination name prefix in the error messages.

File: src/v0/destinations/af/deleteUsers.js (hunk 1 above)
Comment (utsabc) [reply src/v0/destinations/af/deleteUsers.js:49]: mentioned reason in PR description: adding integration names to message to better debug at bugsnag

Comment (saikumarrs) [reply src/v0/destinations/af/deleteUsers.js:49]: Actually, it's not required. The destination name is already captured as part of the metadata and stack trace. Having this prefix pollutes the error messages shown to the user.

File: src/v0/destinations/braze/deleteUsers.js (hunk 2 above)
Comment (utsabc) [reply src/v0/destinations/braze/deleteUsers.js:61]: This is fixed

File: src/v0/destinations/intercom/deleteUsers.js (hunk 3 above)
Comment (utsabc) [reply src/v0/destinations/intercom/deleteUsers.js:36]: This is fixed

File: src/v0/destinations/af/deleteUsers.js (hunk 1 above)
Comment (utsabc) [reply src/v0/destinations/af/deleteUsers.js:49]: Screenshot 2022-12-20 at 6 08 56 PM
Yeah but in bugsnag you have to open individual reports to get that. With this its easier to check the destination from error message. 
I know its not the convention we are aiming for but atleast will help in immediate monitoring later on as deemed we can remove these not a big lift

File: src/v0/destinations/af/deleteUsers.js [hunk 4]
```diff
@@ -89,9 +89,9 @@ const userDeletionHandler = async (userAttributes, config) => {
           if (!body.property_id) {
             throw new ErrorBuilder()
               .setMessage(
-                "appleAppId is required for ios_advertising_id type identifier"
+                "[Appsflyer]::appleAppId is required for ios_advertising_id type identifier"
               )
-              .setStatus(500)
+              .setStatus(400)
```
Comment (saikumarrs) [inline src/v0/destinations/af/deleteUsers.js:94]: Shouldn't we treat configuration errors as retryable?

Comment (utsabc) [reply src/v0/destinations/af/deleteUsers.js:94]: Yes

Comment (utsabc) [reply src/v0/destinations/af/deleteUsers.js]: Fixed

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**